pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -25,7 +25,7 @@ const (
 	kbltCAMountedFile = "ca-bundle.crt"
 	kbltCAName        = "kubelet-ca"
 	defaultScheduler  = "default-scheduler"
-	daemonSetName     = "node-observability-ds"
+	daemonSetName     = "node-observability-agent"
```

Comment:
I'm 100% for this change, it should have been named this way from the beginning.
But I'm a little worried about the migration. What will happen after the new NOB operator version will be rolled out? Will we have 2 daemons? How this will be handled at the profiling time when the operator will need to send a request to each agent?

Reply:
I did this assuming we are ok due to this being in tech-preview. 
https://github.com/openshift/node-observability-operator/pull/90#discussion_r985533378

Maybe we mention in release notes, that we need to uninstall previous and cannot directly upgrade. Seems cleaner than adding migration code for tech-preview.

Reply:
As discussed, let's make this experience as smooth as possible for the customers, especially knowing that the cleanup is feasible and should not take much code. Asking for the reinstall is indeed a measure and we can always use it but I'd prefer to keep it for the case when it's hardly avoidable which is far from this situation.
The rest of the renamings (roles,SCCs) don't have a big impact: no compute resources used, no rights granted. So, I propose to let them rot (=no migration implemented).